docs/extension-e2e-test-guidelines.md

```diff
@@ -0,0 +1,528 @@
+# MetaMask Extension E2E Test Guidelines
+
+## Table of Contents
+
+1. [General](#general)
+2. [Test names](#test-names)
+3. [Organization of test files](#organization-of-test-files)
+4. [Element locators](#element-locators)
+5. [Wait for commands](#wait-for-commands)
+6. [Assertions](#assertions)
+7. [Controlling state](#controlling-state)
+8. [Enhancing test stability with request mocking](#enhancing-test-stability-with-request-mocking)
+9. [Test Atomicity and Smart Test Coupling](#test-atomicity-and-smart-test-coupling)
+10. [Page Object Model](#page-object-model)
```

Comment:
GitHub itself automatically creates a table of contents, so this section might be redundant (and easy to get out of sync in the future)?

Reply:
I updated to use the automatically generated table of content. Thanks David !